### ISSUE TITLE
Receive "Invalid redirect_uri" error when clicking on "Connect to Xero" button

### ISSUE THREAD

**Issue by markabrahams** [XeroAPI/xero-node-oauth2-app#58]
On running this app and connecting to http://localhost:5000 in a browser (Firefox and Chrome tested), I receive the home page fine.  When I click on the "Connect to Xero" button, I receive an error page with text:

Sorry, something went wrong
Go back and try again.
If the issue continues, check out our Status Page.

Error code: 500  Error: unauthorized_client : Invalid redirect_uri

My .env file contains:
CLIENT_ID='<ID-from-app-created-in-Xero>'
CLIENT_SECRET='<secret-from-app-created-in-Xero>'
REDIRECT_URI='http://localhost:5000/callback'

Am running the app on a private address behind a NAT router.  The router forwards TCP port 5000 back to port 5000 on my machine.

URI formats I have tried changing for the value of REDIRECT_URL are (reloading the app after each change):
http://localhost:5000/callback
http://<internal-ip-address>:5000/callback
http://<public-ip-address>:5000/callback
http://<internal-name>:5000/callback
http://<public-name>:5000/callback
https://localhost:5000/callback
https://<internal-ip-address>:5000/callback
https://<public-ip-address>:5000/callback

Tried also with single quotes, double quotes, no quotes, and with or without a trailing empty line.

On a related note, it wasn't clear from the documentation whether the REDIRECT_URL needed to be connectable from Xero, although that was my assumption, hence trying with public addressing and port forwarding.  I verified that the port-forward for TCP port 5000 worked from an external location.

**Comment by RettBehrens** [comment]
Does the redirect url match exactly on both your .env and in the myapps details page in the developer portal? I just ran the sample app on http://localhost:5000 without any extra config regarding port forwarding or public addressing without error. [https://developer.xero.com/documentation/oauth2/troubleshooting](url)

**Comment by markabrahams** [comment]
Hi @RettBehrens  - you're absolutely correct!  My .env and myapps redirect URIs did not match and that was the problem!  Changed these to match, and the app works fine now.  Thanks for your help here and the great speed of your response!

Pretty obvious with hindsight - as many things are!

Suggested improvements that would have helped me and perhaps would help others are these changes to the README.md "Configure with your credentials" section:
- Under the "enter your app details" bullet, for this project, the only valid URI for this project is "http://localhost:5000/callback" - so this should be specific to match the xero-node-oauth2-app project, not generic with variables (it's currently "http://localhost:${PORT}/your-callback-route") as this leaves doubt as to what you might need to change to get it to work (I thought it might be the "localhost" part for example).
- similarly the REDIRECT_URI variable in .env should be shown as "http://localhost:5000/callback".
- a note to say that these need to match or the authentication won't work.  On a re-read, this is strongly implied, but wouldn't hurt to be explicit there to enlighten the more obtuse among us!
- a note to say that the URI does not need to be Internet accessible and that localhost will work fine.  When things went wrong I vortexed down the wrong path, as this wasn't clear and I made the wrong assumption.
